### PR TITLE
feat: Lp/feat/150/entity form wizard

### PR DESCRIPTION
**Pull request by HegelPro**

### CODE REVIEW

File: example-react-app/src/app/form-wizard-CompositionO2O-editor/FormWizardCompositionO2O.tsx [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+          </Form.Item>
+        </Form>
+      </Card>
+    );
+  }
+);
+
+registerEntityEditor({
+  component: FormWizardCompositionO2O,
+  caption: "screen.FormWizardCompositionO2O",
```
Comment (vyacheslav-pushkin) [inline example-react-app/src/app/form-wizard-CompositionO2O-editor/FormWizardCompositionO2O.tsx:130]: Submitting the form results in:

image

(even though GraphQL mutation seems to be successful)

Also editor doesn't close (should it?).

File: example-react-app/src/app/form-wizard-CompositionO2O-editor/FormWizardCompositionO2O.tsx [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+  screenId: "FormWizardCompositionO2O",
```
Comment (vyacheslav-pushkin) [inline example-react-app/src/app/form-wizard-CompositionO2O-editor/FormWizardCompositionO2O.tsx:131]: Clicking `Cancel` seems to cause a crash:

```
index.esm.js:8148 Uncaught TypeError: Cannot read property 'params' of null
    at Screens.getUrl (index.esm.js:8148)
    at Screens.<anonymous> (index.esm.js:8107)
    at executeAction (action.ts:65)
    at Screens.<unnamed action> [as setActiveScreen] (action.ts:46)
    at index.esm.js:572
    at handleClick (button.js:207)
    at HTMLUnknownElement.callCallback (react-dom.development.js:3945)
    at Object.invokeGuardedCallbackDev (react-dom.development.js:3994)
    at invokeGuardedCallback (react-dom.development.js:4056)
    at invokeGuardedCallbackAndCatchFirstError (react-dom.development.js:4070)
    at executeDispatch (react-dom.development.js:8243)
    at processDispatchQueueItemsInOrder (react-dom.development.js:8275)
    at processDispatchQueue (react-dom.development.js:8288)
    at dispatchEventsForPlugins (react-dom.development.js:8299)
    at react-dom.development.js:8508
    at batchedEventUpdates$1 (react-dom.development.js:22396)
    at batchedEventUpdates (react-dom.development.js:3745)
    at dispatchEventForPluginEventSystem (react-dom.development.js:8507)
    at attemptToDispatchEvent (react-dom.development.js:6005)
    at dispatchEvent (react-dom.development.js:5924)
    at unstable_runWithPriority (scheduler.development.js:468)
    at runWithPriority$1 (react-dom.development.js:11276)
    at discreteUpdates$1 (react-dom.development.js:22413)
    at discreteUpdates (react-dom.development.js:3756)
    at dispatchDiscreteEvent (react-dom.development.js:5889)
```

File: example-react-app/src/app/form-wizard-CompositionO2O-editor/FormWizardCompositionO2O.tsx (hunk 1 above)
Comment (vyacheslav-pushkin) [reply example-react-app/src/app/form-wizard-CompositionO2O-editor/FormWizardCompositionO2O.tsx:130]: Only happens in `Form Wizard Editor` screen.

File: example-react-app/src/app/form-wizard-CompositionO2O-editor/FormWizardCompositionO2O.tsx (hunk 2 above)
Comment (vyacheslav-pushkin) [reply example-react-app/src/app/form-wizard-CompositionO2O-editor/FormWizardCompositionO2O.tsx:131]: Also only happens in `Form Wizard Editor`.

Comment (vyacheslav-pushkin) [reply example-react-app/src/app/form-wizard-CompositionO2O-editor/FormWizardCompositionO2O.tsx:131]: Both issues are likely to be unrelated to entity wizard itself and related to screens API. Probably its best to create a separate issue for that.

File: packages/jmix-front-generator/src/generators/react-typescript/entity-form-wizard/template-model.ts [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+  const {
+    associations,
+    compositions,
+    nestedEntityInfo,
+    readOnlyFields,
+    relationImports,
+    associatedEntityClassNames
+  } = deriveEditorTemplateModel(answers, projectModel, displayedAttributes)
+  // const {editAssociatedEntityClassNames} = deriveEditAssociatedEntityClassNames(associations)
```
Comment (vyacheslav-pushkin) [inline packages/jmix-front-generator/src/generators/react-typescript/entity-form-wizard/template-model.ts:53]: Why do we have it commented-out?

File: packages/jmix-react-ui/src/ui/FormWizard/FormWizardContext.tsx [hunk 4]
```diff
@@ -0,0 +1,55 @@
+import React, {useContext, useState, createContext} from 'react';
+import { FormWizardStore } from './FormWizardStore';
+
+interface FormWizardHelpers {
+    validateFields: () => Promise<void>;
+    onFinish: (values: any) => void;
+    onFinishFailed?: (error: any) => void;
+} 
+
+const defaultHerlpersRef: React.MutableRefObject<FormWizardHelpers> = {
```
Comment (vyacheslav-pushkin) [inline packages/jmix-react-ui/src/ui/FormWizard/FormWizardContext.tsx:10]: Typo in `Herlpers`

File: packages/jmix-react-ui/src/ui/FormWizard/useEntityEditorFormWizard.tsx [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+    const handleSelectStep = useCallback(async newStepIndex => {
+        await formWizardHelpersRef.current.validateFields();
+        formWizardStore.setStepIndex(newStepIndex);
+    }, [formWizardStore, formWizardHelpersRef.current.validateFields]);
+    
+    const handleNextStep = useCallback(async () => {
+        await formWizardHelpersRef.current.validateFields();
+        formWizardStore.next();
+    }, [formWizardStore, formWizardHelpersRef.current, formWizardHelpersRef.current.validateFields]);
+
```
Comment (vyacheslav-pushkin) [inline packages/jmix-react-ui/src/ui/FormWizard/useEntityEditorFormWizard.tsx:54]: I think we should not proceed to next step if there are validation errors in the current step.

File: packages/jmix-react-ui/src/ui/FormWizard/FormWizardSteper.tsx [hunk 6]
```diff
@@ -0,0 +1,29 @@
+import { observer } from "mobx-react";
+import React from "react";
+import { useFormWizard } from "./FormWizardContext";
+import {Steps} from 'antd';
+
+interface FormWizardSteperProps {
+    onSelectStep: (stepIndex: number) => void;
+}
+
+export const FormWizardSteper = observer(({onSelectStep}: FormWizardSteperProps) => {
```
Comment (vyacheslav-pushkin) [inline packages/jmix-react-ui/src/ui/FormWizard/FormWizardSteper.tsx:10]: I think we need a different name here. `FormWizardSteps` maybe?

File: packages/jmix-react-ui/src/ui/FormWizard/FormWizardStore.ts [hunk 7]
```diff
@@ -0,0 +1,153 @@
...
+    // setFieldErrorsForSteps(fieldErrors: Map<string, string[]>) {
+    //     this.steps.forEach(step => {
+    //         step.setServerFieldErrors(
+    //             selectStepFieldErrors(fieldErrors, step.FIELD_NAMES)
+    //         )
+    //         if(step.serverFieldErrors.size > 0) {
+    //             step.setStatus('error')
+    //         }
+    //     })
+    // }
```
Comment (vyacheslav-pushkin) [inline packages/jmix-react-ui/src/ui/FormWizard/FormWizardStore.ts:135]: Why do we have commented-out code?

File: packages/jmix-front-generator/src/test/generators/react-typrscript/common/i18n.test.ts [hunk 8]
```diff
@@ -216,6 +216,8 @@ const ALL_I18N_KEYS = [
   "common.edit",
   "common.remove",
   "common.submit",
+  "common.next",
+  "common.previous",
```
Comment (vyacheslav-pushkin) [inline packages/jmix-front-generator/src/test/generators/react-typrscript/common/i18n.test.ts:220]: Looks like these keys a missing in `fr` and `zh-cn` files.

**Review by vyacheslav-pushkin — CHANGES_REQUESTED** [review]

File: example-react-app/src/app/form-wizard-CompositionO2O-editor/FormWizardCompositionO2O.tsx (hunk 2 above)
Comment (HegelPro) [reply example-react-app/src/app/form-wizard-CompositionO2O-editor/FormWizardCompositionO2O.tsx:131]: Yes, it doesn't work for other editors either. I created an issue:
https://github.com/Haulmont/jmix-frontend/issues/509

File: example-react-app/src/app/form-wizard-CompositionO2O-editor/FormWizardCompositionO2O.tsx (hunk 1 above)
Comment (HegelPro) [reply example-react-app/src/app/form-wizard-CompositionO2O-editor/FormWizardCompositionO2O.tsx:130]: Created the issue for it.
https://github.com/Haulmont/jmix-frontend/issues/510

**Review by vyacheslav-pushkin — APPROVED** [review]

**Pull request merged**